src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,121 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int threshold = Math.round(DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        int index = getIndex(key);
+        Node<K, V> processingNode = table[index];
+        if (processingNode == null) {
+            table[index] = new Node<K, V>(key, value, null);
+            size++;
+        } else {
+            while (processingNode.next != null) {
+                if (areValuesEqual(processingNode.key, key)) {
+                    processingNode.value = value;
+                    return;
+                }
+                processingNode = processingNode.next;
+            }
+            if (areValuesEqual(processingNode.key, key)) {
+                processingNode.value = value;
+                return;
+            }
+            processingNode.next = new Node(key, value, null);
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int index = getIndex(key);
+        Node<K, V> processingNode = table[index];
+        while (true) {
```

Comment:
It is possible to use loop similar to loop that you used in resize method, such approach is more readable